### PR TITLE
Refactor distributed graph store

### PR DESCRIPTION
**Pull request by zheng-da**
## Description
Refactor the distributed graph store for the new kvstore.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the my best knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR

### CODE REVIEW

File: python/dgl/distributed/graph_partition_book.py [hunk 1]
```diff
@@ -78,6 +141,18 @@ def __init__(self, part_id, num_parts, node_map, edge_map, part_graph):
...
+
+        Parameters
+        ----------
+        graph_name : str
+            The graph name
+        """
+        self._meta, self._nid2partid, self._eid2partid = _move_metadata_to_shared_mam(
+            graph_name, self.num_nodes(), self.num_edges(), self._part_id, self._num_partitions,
+            self._nid2partid, self._eid2partid)
+
```
Comment (aksnzhy) [inline python/dgl/distributed/graph_partition_book.py:155]: So we will not put GraphPartitionBook to ServerState, right?

File: python/dgl/distributed/kvstore.py [hunk 2]
```diff
@@ -953,7 +935,6 @@ def map_shared_data(self, partition_book):
             for _ in range(self._group_count-1):
                 response = rpc.recv_response()
                 assert response.msg == SEND_META_TO_BACKUP_MSG
-        self._freeze = True
```
Comment (aksnzhy) [inline python/dgl/distributed/kvstore.py:956]: I think we cannot just simply remove the `freeze` code here. I will change the code in a new PR to fix it.

**Comment by aksnzhy** [comment]
The overall comment is: should we put all the data (except for the dynamic data tensor in kvstore) like GraphPartitionBook and meta data to ServerState and manage the shared-memory in a unified way?

**Comment by zheng-da** [comment]
I, in general, prefer not to. If we want to put everything in server state, it makes the design inflexible to a future extension. If we want to add something, we need to modify the server state. what is the benefit of having everything managed by server state?

File: python/dgl/distributed/graph_partition_book.py (hunk 1 above)
Comment (zheng-da) [reply python/dgl/distributed/graph_partition_book.py:155]: we can have graph partition book referenced by server state. but we don't need to have it managed by server state.

**Comment by aksnzhy** [comment]
I removed the `freeze` in this pr:https://github.com/dmlc/dgl/pull/1605, and I solved the conflict in this PR

**Comment by zheng-da** [comment]
i have added the reference of graph partition book to server state.

**Pull request merged**